core/stats.go

```diff
@@ -1,7 +1,42 @@
 package core
 
-var KeyspaceStat [4]map[string]int
+import (
+	"sync"
+)
 
-func UpdateDBStat(num int, metric string, value int) {
-	KeyspaceStat[num][metric] = value
+type KeyspaceStat struct {
+	stats map[string]int
+	mutex sync.Mutex
```

Comment:
@soumya-codes do you think its okay for us to use this in-built lock for managing the KeyspaceStat? Other mutexes use the functional pattern defined in the locks.go file.

I don't have strong opinions on this. The proposed method in this PR looks good and concise for the use case, however, it deviates from the existing pattern.

Reply:
IMO it is good to follow the same pattern across the code. 
Also, In this case, it appears we will not take any additional performance hit by adding a lock over the whole operation like we do with other lock types.